frontend/src/scenes/dashboard/dashboards/dashboardsLogic.ts

```diff
@@ -0,0 +1,100 @@
+import { kea } from 'kea'
+import Fuse from 'fuse.js'
+import { dashboardsModel } from '~/models/dashboardsModel'
+import type { dashboardsLogicType } from './dashboardsLogicType'
+import { DashboardType } from '~/types'
+import { uniqueBy } from 'lib/utils'
+import { userLogic } from 'scenes/userLogic'
+import { dashboardTemplateLogic } from 'scenes/dashboard/dashboardTemplates/dashboardTemplateLogic'
+
+export enum DashboardsTab {
+    All = 'all',
+    Yours = 'yours',
+    Pinned = 'pinned',
+    Shared = 'shared',
+    Templates = 'templates',
+}
+
+export const dashboardsLogic = kea<dashboardsLogicType>({
+    path: ['scenes', 'dashboard', 'dashboardsLogic'],
+    connect: {
+        values: [userLogic, ['user'], dashboardTemplateLogic, ['dashboardTemplates']],
+        actions: [dashboardTemplateLogic, ['getAllDashboardTemplates']],
+    },
+    actions: {
+        setSearchTerm: (searchTerm: string) => ({ searchTerm }),
+        setCurrentTab: (tab: DashboardsTab) => ({ tab }),
+    },
+    reducers: {
+        searchTerm: {
+            setSearchTerm: (_, { searchTerm }) => searchTerm,
+        },
+        currentTab: [
+            DashboardsTab.All as DashboardsTab,
+            {
+                setCurrentTab: (_, { tab }) => tab,
+            },
+        ],
+    },
+    selectors: {
+        filteredDashboardTemplates: [
```

Comment:
GitHub really needs semantic diff. This file shows as all added but it's only this selector that has been added